c++/src/kj/filesystem-disk-test.c++

```diff
@@ -945,5 +946,30 @@ KJ_TEST("DiskFile holes") {
 }
 #endif
 
+// Ensure the current path is correctly computed.
+//
+// See issue #1425.
+KJ_TEST("DiskFilesystem::computeCurrentPath") {
+  TempDir tempDir;
+  auto dir = tempDir.get();
+
+  // Paths can be PATH_MAX, but the segments which make up that path typically
+  // can't exceed 255 bytes.
+  auto maxPathSegment = std::string(255, 'a');
+
+  // Create a path which exceeds the 256 byte buffer used in
+  // computeCurrentPath. 
+  auto subdir = dir->openSubdir(Path({
+    maxPathSegment,
+    maxPathSegment,
+    "some_path_longer_than_256_bytes"
```

Comment:
Running the CI now and we'll see what it says, but I'm going to guess this part fails on Windows and maybe some other OSes. Might need to limit to Linux.

Reply:
Interesting thought.

I've pushed a change which addresses the unused result errors. Let's see if CI is any happier now, and if not I'm happy to restrict this to Linux.

Reply:
It actually does appear that it passed on windows-latest so maybe Windows does support long paths after all? Somehow I thought it was limited to 256 chars total for the whole path.

Actually I bet it works specifically because the KJ filesystem impl uses the crazy `\\?\` prefix that bypasses a huge amount of Win32 legacy cruft, heh.

Reply:
Wait no. The windows-latest build is broken and is not even doing a build, but reporting success. :man_facepalming:

Reply:
😅 Restricted the test to Unix.